diff --git a/_build/default/lib/eleveldb/Makefile b/_build/default/lib/eleveldb/Makefile
index b0d0068..98a0fe7 100644
--- a/_build/default/lib/eleveldb/Makefile
+++ b/_build/default/lib/eleveldb/Makefile
@@ -1,12 +1,9 @@
-.PHONY: compile rel cover test dialyzer get-deps
+.PHONY: compile rel cover test dialyzer
 REBAR=./rebar3
 
 all: compile
 
-get-deps:
-	$(REBAR) get-deps
-
-compile: get-deps
+compile:
 	$(REBAR) compile
 
 clean:
diff --git a/_build/default/lib/eleveldb/c_src/Makefile b/_build/default/lib/eleveldb/c_src/Makefile
index 8de7e33..b9b1b46 100644
--- a/_build/default/lib/eleveldb/c_src/Makefile
+++ b/_build/default/lib/eleveldb/c_src/Makefile
@@ -21,8 +21,8 @@ compile: get-deps snappy ldb
 	cp leveldb/perf_dump leveldb/sst_rewrite leveldb/sst_scan leveldb/leveldb_repair ../priv
 
 ldb:
-	$(MAKE) LDFLAGS="$(LDFLAGS) -lsnappy" LD_LIBRARY_PATH="$(LD_LIBRARY_PATH)" -C leveldb all
-	$(MAKE) LDFLAGS="$(LDFLAGS) -lsnappy" LD_LIBRARY_PATH="$(LD_LIBRARY_PATH)" -C leveldb tools
+	$(MAKE) LDFLAGS="$(LDFLAGS) -lsnappy -lpthread" LD_LIBRARY_PATH="$(LD_LIBRARY_PATH)" -C leveldb all
+	$(MAKE) LDFLAGS="$(LDFLAGS) -lsnappy -lpthread" LD_LIBRARY_PATH="$(LD_LIBRARY_PATH)" -C leveldb tools
 
 snappy: system/lib/libsnappy.a
 
@@ -31,7 +31,7 @@ system/lib/libsnappy.a:
 	(cd snappy-$(SNAPPY_VSN) && \
 	 git submodule update --init && \
 	 if [ -r autogen.sh ]; then \
-	   ./autogen.sh && ./configure --prefix=$(BASEDIR)/system && make && make install; \
+	   ./autogen.sh && ./configure --prefix=$(BASEDIR)/system && $(MAKE) && $(MAKE) install; \
 	 else \
 	   mkdir build && cd build && \
 	   mkdir -p $(BASEDIR)/system && \
@@ -39,7 +39,7 @@ system/lib/libsnappy.a:
 	         -D CMAKE_INSTALL_PREFIX=$(BASEDIR)/system \
 	     ..; \
 	  fi && \
-	  make && make install)
+	  $(MAKE) && $(MAKE) install)
 	mv system/lib64 system/lib || true
 
 clean:
diff --git a/_build/default/lib/eleveldb/make b/_build/default/lib/eleveldb/make
new file mode 100755
index 0000000..fcda4c3
--- /dev/null
+++ b/_build/default/lib/eleveldb/make
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+which gmake 1>/dev/null 2>/dev/null && MAKE=gmake
+MAKE=${MAKE:-make}
+
+$MAKE $@
diff --git a/_build/default/lib/eleveldb/rebar.config b/_build/default/lib/eleveldb/rebar.config
index 200b16d..8ec3579 100644
--- a/_build/default/lib/eleveldb/rebar.config
+++ b/_build/default/lib/eleveldb/rebar.config
@@ -36,7 +36,7 @@
              {"DRV_LDFLAGS", "$DRV_LDFLAGS c_src/leveldb/libleveldb.a c_src/system/lib/libsnappy.a -lstdc++"}
              ]}.
 
-{pre_hooks, [{'get-deps', "make -C c_src get-deps"},
-             {compile, "make -C c_src compile"}]}.
+{pre_hooks, [{'get-deps', "./make -C c_src get-deps"},
+             {compile, "./make -C c_src compile"}]}.
 
-{post_hooks, [{clean, "make -C c_src clean"}]}.
+{post_hooks, [{clean, "./make -C c_src clean"}]}.
